### ISSUE TITLE
jcabi-ssl-maven-plugin doesn't work when building on Rultor or Travis

### ISSUE THREAD

**Issue by carlosmiranda** [jcabi/jcabi-ssl-maven-plugin#2]
I'm trying to enable a SSL-related unit test in the S3Auth project (see https://github.com/yegor256/s3auth/pull/194). I incorporated `jcabi-ssl-maven-plugin` in the Maven build, and the unit test passes just fine on my machine. However, the build fails on Rultor. See for example, the following log excerpt (from http://www.rultor.com/t/529-49795456):

```
Tests run: 10, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 322.974 sec <<< FAILURE! - in com.s3auth.relay.HttpFacadeTest
getsContentOverSSL(com.s3auth.relay.HttpFacadeTest) Time elapsed: 31.773 sec <<< ERROR! java.io.IOException: Failed GET request to https://localhost:59055/a at com.jcabi.http.request.JdkRequest$1.send(JdkRequest.java:125) at
com.jcabi.http.request.BaseRequest.fetch_aroundBody10(BaseRequest.java:210) at 
com.jcabi.http.request.BaseRequest$AjcClosure11.run(BaseRequest.java:1) at 
org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) at 
com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:200) at 
com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1) at 
com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:128) at 
com.jcabi.http.request.BaseRequest.fetch(BaseRequest.java:209) at 
com.s3auth.relay.HttpFacadeTest.getsContentOverSSL(HttpFacadeTest.java:533) at 
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at 
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) at 
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at 
java.lang.reflect.Method.invoke(Method.java:606) at 
org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) at 
org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) at 
org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) at 
org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) at 
org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) at 
org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) at 
org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) at 
org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) at 
org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:258) at 
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471) at 
java.util.concurrent.FutureTask.run(FutureTask.java:262) at 
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) at 
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) at 
java.lang.Thread.run(Thread.java:744)
Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) at 
sun.security.ssl.Alerts.getSSLException(Alerts.java:154) at 
sun.security.ssl.SSLSocketImpl.recvAlert(SSLSocketImpl.java:1959) at 
sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1077) at 
sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1312) at 
sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1339) at 
sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1323) at 
sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:563) at 
sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185) at 
sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1300) at 
java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468) at 
sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338) at
com.jcabi.http.request.JdkRequest$1.send(JdkRequest.java:117) ... 26 more
```

If it helps, this is my machine's `mvn --version` output:
```
Apache Maven 3.2.1 (ea8b2b07643dbb1b84b6d16e1f08391b666bc1e9; 2014-02-15T01:37:52+08:00)
Maven home: D:\apache-maven-3.2.1
Java version: 1.7.0_45, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk1.7.0_45\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 7", version: "6.1", arch: "x86", family: "windows"
```

**Comment by dmarkov** [comment]
@cyberone you're holding this ticket for 13 days already. If it's not finished (and closed) in the next 16 hours, it will be re-assigned to someone else, see [No Obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle... -30 added to your rating, current score is: [-220](http://www.netbout.com/b/35076?open=rating)

**Comment by dmarkov** [comment]
@cyberone it takes too long. Usually, we expect any task to be finished in less than a week. I'll assign someone else. Please stop working with it right now. See our [no obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle

-60 added to your rating, at the moment it is: [-430](http://www.netbout.com/b/35076?open=rating)

**Comment by dmarkov** [comment]
@aoxa please go ahead, it's your task now, keep [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind, and don't hesitate to ask any technical questions you may have

The budget here is **30 mins**, which is exactly how much time will be paid for, when the task is [completed](http://www.xdsd.org/2014/04/15/definition-of-done.html)

**Comment by dmarkov** [comment]
@aoxa the ticket is with you for 15 days already. If it's not finished in the next 48 hours, it will be re-assigned to someone else, see [No Obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle, [this article](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html) should help if you're stuck. added -30 to your rating, now it is equal to -90

**Comment by dmarkov** [comment]
@aoxa the task is overdue, and I have to re-assign it to someone else. Please, stop working with it immediately. In general, we're against overdue tasks, check [this page](http://www.xdsd.org/2014/04/13/no-obligations-principle.html). added -60 to your rating, now it is equal to -570

**Comment by dmarkov** [comment]
@suseika it's yours, budget is 30 mins, please go ahead

**Comment by Suseika** [comment]
@dmarkov Please assign someone else on this issue

**Comment by dmarkov** [comment]
> @dmarkov Please assign someone else on this issue

@suseika 30 points deducted from your rating

**Comment by dmarkov** [comment]
> @dmarkov Please assign someone else on this issue

@suseika no problem, I will try to find somebody else

**Comment by dmarkov** [comment]
@prondzyn this task is for you, keep [these principles](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind, and don't hesitate to ask any technical questions... This task's budget is **30 mins**. This is exactly how much will be paid when the problem explained above is solved. See [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for more information

**Comment by prondzyn** [comment]
@carlosmiranda this test does not pass on my machine:

```
Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 347.017 sec <<< FAILURE! - in com.s3auth.relay.HttpFacadeTest
getsContentOverSSL(com.s3auth.relay.HttpFacadeTest)  Time elapsed: 31.156 sec  <<< FAILURE!
java.lang.AssertionError: HTTP response body content is not valid:
200 OK [https://localhost:61301/a]
Age: 0
Date: Pt, 16 pa?? 2015 08:20:58 +0200
X-S3auth-Time: 128
Server: relay.s3auth.com, 1.0-SNAPSHOT/123 built on 22-Aug-2012

<<empty>>
Expected: is "secured"
     but: was ""
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at com.jcabi.http.response.RestResponse.assertBody(RestResponse.java:158)
        at com.s3auth.relay.HttpFacadeTest.getsContentOverSSL(HttpFacadeTest.java:555)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:318)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

but you wrote in [194](https://github.com/yegor256/s3auth/pull/194) that it always works when you try to build.

My environment:
```
$ mvn -version
Apache Maven 3.2.2 (45f7c06d68e745d05611f7fd14efb6594181933e; 2014-06-17T15:51:42+02:00)
Maven home: C:\local\tools\apache-maven-3.2.2
Java version: 1.7.0_60, vendor: Oracle Corporation
Java home: C:\opt\java\jdk1.7.0\jre
Default locale: pl_PL, platform encoding: Cp1250
OS name: "windows 7", version: "6.1", arch: "x86", family: "windows"
```

Could you remove the `@Ignore` annotation and try to build the latest version from [yegor256/s3auth](https://github.com/yegor256/s3auth)? Should I prepare a separate issue for this?

**Comment by prondzyn** [comment]
@dmarkov I reported the issue in [yegor256/s3auth/issues/313](https://github.com/yegor256/s3auth/issues/313) as this is blocking my work on current issue.

**Comment by prondzyn** [comment]
@dmarkov after [conversation](https://github.com/yegor256/s3auth/issues/313) I found that there is already puzzle reported for this in [yegor256/s3auth/issues/290](https://github.com/yegor256/s3auth/issues/290). 

So I am blocked by [290](https://github.com/yegor256/s3auth/issues/290).

**Comment by dmarkov** [comment]
> @dmarkov I reported the issue in [yegor256/s3auth/issues/313](https://github.com/yegor256/s3auth/issues/313) as this is blocking my work on current issue.

@prondzyn of course, take your time

**Comment by prondzyn** [comment]
@dmarkov I'm away now. Maybe you should find someone else for this?

**Comment by dmarkov** [comment]
> @dmarkov I'm away now. Maybe you should find someone else for this?

@prondzyn deducted 30 from your rating :(

**Comment by dmarkov** [comment]
> @dmarkov I'm away now. Maybe you should find someone else for this?

@prondzyn no problem, I will try to find somebody else

**Comment by dmarkov** [comment]
@amihaiemil it's your task, go ahead

**Comment by amihaiemil** [comment]
@carlosmiranda Is this issue still valid? It's been almost 2 years ago, maybe it was fixed in the meantime?

**Comment by amihaiemil** [comment]
@carlosmiranda from my understanding the issue will be solved [this](https://github.com/yegor256/s3auth/issues/290) ticket?

**Comment by carlosmiranda** [comment]
@amihaiemil Actually, it looks like it was fixed in #8, which was applied to S3Auth in https://github.com/yegor256/s3auth/pull/238.

Thanks for looking at it - closing now.

**Comment by dmarkov** [comment]
@amihaiemil thanks, I just added **1 hour** to your account, payment `AP-0JM36599MS278841G`, 19 hours and 48 mins spent... there is a bonus for fast delivery (m=1188)... +60 added to your rating, current score is: [+225](http://www.netbout.com/b/36402?open=rating)
